tests/unit/s2n_build_test.c

```diff
@@ -0,0 +1,228 @@
+    if (s2n_build_preset == NULL && s2n_libcrypto == NULL) {
+        /* No intent declared. */
+        END_TEST();
+        return 0;
+    }
+
+    /* Parse the info out of environment variables.  */
+    struct libcrypto_info s2n_build_preset_info = { 0 };
+    init_libcrypto_info(&s2n_build_preset_info);
+    char s2n_build_preset_copy[100] = { 0 };
+    int s2n_build_preset_parsed = -1;
+    if (s2n_build_preset != NULL && s2n_build_preset[0] != '\0') {
+        strcpy(s2n_build_preset_copy, s2n_build_preset);
+        s2n_build_preset_parsed = extract_libcrypto_info_from_s2n_build_preset(s2n_build_preset_copy, &s2n_build_preset_info);
+    }
+
+    struct libcrypto_info s2n_libcrypto_info = { 0 };
+    init_libcrypto_info(&s2n_libcrypto_info);
+    char s2n_libcrypto_copy[100] = { 0 };
+    int s2n_libcrypto_parsed = -1;
+    if (s2n_libcrypto != NULL && s2n_libcrypto[0] != '\0') {
+        strcpy(s2n_libcrypto_copy, s2n_libcrypto);
+        s2n_libcrypto_parsed = extract_libcrypto_info_from_s2n_libcrypto(s2n_libcrypto_copy, &s2n_libcrypto_info);
+    }
+
+    if (s2n_libcrypto_parsed == -1 && s2n_build_preset_parsed == -1) {
+        /* Neither intent parsed.  */
+        END_TEST();
+        return 0;
+    }
+
+    /* If S2N_BUILD_PRESET and S2N_LIBCRYPTO don't match the intent of the CI is inconsistent.  */
+    if (s2n_libcrypto_parsed != -1 && s2n_build_preset_parsed != -1) {
+        /* Check that intents, if both declared, match.  */
+        EXPECT_STRING_EQUAL(s2n_libcrypto_info.name, s2n_build_preset_info.name);
+        EXPECT_EQUAL(s2n_libcrypto_info.major_version, s2n_build_preset_info.major_version);
+        EXPECT_EQUAL(s2n_libcrypto_info.minor_version, s2n_build_preset_info.minor_version);
+        EXPECT_EQUAL(s2n_libcrypto_info.patch_version, s2n_build_preset_info.patch_version);
+        EXPECT_EQUAL(s2n_libcrypto_info.is_fips, s2n_build_preset_info.is_fips);
+    }
+
+    /* The intent of the CI is defined and consistent.  */
+    struct libcrypto_info *s2n_libcrypto_intent = NULL;
+    if (s2n_libcrypto_parsed != -1) {
+        s2n_libcrypto_intent = &s2n_libcrypto_info;
+    } else if (s2n_build_preset_parsed != -1) {
+        s2n_libcrypto_intent = &s2n_build_preset_info;
+    } else {
+        FAIL_MSG("Sanity check -- this should be unreachable");
+    }
+
+    /* Check libcrypto name matches the intent of the CI.  */
+    {
+        if (strcmp(s2n_libcrypto_intent->name, "awslc") == 0) {
+            /* aws lc version strings have a storied history:
+             *     Originally a fork from BoringSSL, AWSLC's OPENSSL_VERSION_TEXT was defined to be "OpenSSL 1.1.1 (compatible; BoringSSL)".
+             *     AWC-LC commit 8f184f5d69604cc4645bafec47c2d6d9929cb50f on 4/11/22 modified it to be  "OpenSSL 1.1.1 (compatible; AWL-LC)"
+             * Unfortunately in both cases we can't do a straight forward comparison against "S2N_LIBCRYPTO=awslc".
+             * We can however rely on the macro OPENSSL_IS_AWSLC.  */
```

Comment:
I think you can just say:
/*  Special case for AWS-LC since it's a fork of BoringSSL and therefore the version returned by SSLeay_version is inaccurate. */
Also I think you should align the comment with the thing you're trying to explain, so in this case, the #ifndef.

Reply:
Should we open a ticket with AWS-LC to get them to fix their SSLeay_version return value 😆 ?

Reply:
Do you mean change the value in a old release? --> I think that would be a breaking api change on a FIPS validated version of the library. But we can try.

Reply:
Interestingly enough, this PR was just opened from the AWS-LC folks...https://github.com/aws/s2n-tls/pull/3779 I wonder if this will affect your PR 🤔